dev/io.openliberty.io.smallrye.reactive.messaging-provider4/src/io/openliberty/microprofile/reactive/messaging30/internal/OLReactiveMessaging30Extension.java

```diff
@@ -87,6 +95,23 @@ void beforeBeanDiscovery(@Observes BeforeBeanDiscovery discovery, BeanManager be
         addQualifier(Connector.class, discovery, beanManager);
     }
 
+    // Set priority to be higher then default, so will run after the ReactiveMessagingExtension event
+    void postDeploymentValidation(@Observes @Priority(jakarta.interceptor.Interceptor.Priority.APPLICATION + 1000) AfterDeploymentValidation done, BeanManager beanManager) {
```

Comment:
If you change this method signature so it overrides `protected void afterDeploymentValidation(@Observes AfterDeploymentValidation done, BeanManager beanManager)` from the superclass and add `MediatorManager mediatorManager = configureMediatorManager(beanManager);` to the start of the try block, I think you can remove the overlay entirely.